src/partition/repart.c

```diff
@@ -2455,6 +2455,20 @@ static int partition_read_definition(Partition *p, const char *path, const char
                     PARTITION_USR_VERITY) || p->verity == VERITY_DATA) && p->read_only < 0)
                 p->read_only = true;
 
+        /* Don't automatically grow ESP, XBOOTLDR and verity partitions. ESP and XBOOTLDR are*/
+        if (p->weight == UINT32_MAX) {
+                if (IN_SET(p->type.designator,
+                           PARTITION_ROOT_VERITY,
+                           PARTITION_USR_VERITY,
+                           PARTITION_ROOT_VERITY_SIG,
+                           PARTITION_USR_VERITY_SIG,
+                           PARTITION_ESP,
+                           PARTITION_XBOOTLDR) || p->verity == VERITY_DATA)
```

Comment:
Why only `p->verity == VERITY_DATA` ? Is `VERITY_SIG` covered by the type designator ?

Reply:
It's messy, they're somewhat distinct concepts, I'll keep to the designators only for now